### ISSUE TITLE
send_scouting function: Iterator restart broken

### ISSUE THREAD

**Issue by tfc** [rosenpass/rosenpass#448]
Hi,

in function `send_scouting` in file https://github.com/rosenpass/rosenpass/blob/de51c1005f476d15f242fce9d9903245307b718f/rosenpass/src/app_server.rs#L493, it can happen that the iterator of the inner loop appears empty if a foregoing iteration of the outer loop resulted in an `Err(...)` result in the inner loop.

I triggered this by setting the listen address (eronneously) to `[::1]:<port>` in my rosenpass config.
(Everything works with `[::]:<port>` as listen address, but this was still considered a bug by @koraa in our conversation today)

It appears that the "iterator restart" is broken, as `collect`ing into a new `Vec` fixes the problem.

**Comment by aparcar** [comment]
Kindly have a look at https://github.com/rosenpass/rosenpass/pull/465 and see if it fixes the issue.

While this seems to work for me, I'm still a bit unhappy with the handling of dual stack. The Linux distribution I tried would not allow to have IPv6 dual stack at all, on macOS it's reported as dual stack but then fails to connect.

If this is a security consideration, i.e. we only open sockets on interface we want to connect with (say 10.0.0.0/16 instead of Internet facing interfaces), then the silent "upgrade" to an empty "listen" list to use `[..]` per default seems problematic.

In other words, why do we do the round robin approach at all and don't always use `[..]`? Right now the ability to connect partly depends on what the user defines in the `listen` option which seems counter intuitive.
